nova/compute/resource_tracker.py

```diff
@@ -1292,15 +1293,22 @@ def _update_usage_from_instances(self, context, instances, nodename):
                 "auto-correction of allocations.")
 
         instance_by_uuid = {}
+        for instance in instances:
+            instance_by_uuid[instance.uuid] = instance
+
+        bdms_by_uuid = (
```

Comment:
The name is confusing, as bdms have a uuid, too, and we have `instance_by_uuid` above following the same concept. It should be something like `bdms_by_instance_uuid`.